src/main/java/com/amplitude/Amplitude.java

```diff
@@ -24,6 +24,7 @@ public class Amplitude {
   private Object eventQueueLock = new Object();
   private Plan plan;
   private long flushTimeout;
+  private String libraryContext;
```

Comment:
I found that adding a new middleware to this repo might be a little overkilled, e.g.:
```java
public class ContextMiddleware implements Middleware {
    public String libraryContext;

    public ContextMiddleware(String libraryContext) {
        this.libraryContext = libraryContext;
    }

    @Override
    public void run(MiddlewarePayload payload, MiddlewareNext next) {
        if (payload.event != null) {
            payload.event.libraryContext = this.libraryContext;
        }
        next.run(payload);
    }
}
```

Reply:
What'll be your suggestion?

Reply:
Oh, my preference would be the change in this PR.
- The property can be globally set (like `plan`), doesn't need to be applied per event
- Lightweight and minimum changes, adding a new middleware and exporting it in package only for this case seems overkill to me

For the Python (new generation SDk) change: https://github.com/amplitude/Amplitude-Python/pull/33/files. Similarly, exposing a way to set it globally, as Python puts the `library` in ContextPlugin, we can put in the similar place, while keeping the changes minimum and lightweight.

Reply:
Got it! Thanks